openmdao/core/group.py

```diff
@@ -1740,6 +1767,21 @@ def _linearize(self, jac, sub_do_ln=True):
                     if subsys._linear_solver is not None:
                         subsys._linear_solver._linearize()
 
+    def _check_first_linearize(self):
+        if self._first_call_to_linearize:
```

Comment:
First 7 lines are duplicated in component.  Could put them in system instead?

Reply:
I decided not to do that in this case because the differences are inside of conditionals and I would've had to either repeat the conditionals or do some return value thing that didn't seem as clean.